src/object_async/hello_async.cpp

```diff
@@ -165,8 +168,11 @@ struct AsyncHelloWorker : Nan::AsyncWorker
 {
 
     using Base = Nan::AsyncWorker;
+    // Make this class noncopyable
+    AsyncHelloWorker(AsyncHelloWorker const&) = delete;
+    AsyncHelloWorker& operator=(AsyncHelloWorker const&) = delete;
 
-    AsyncHelloWorker(bool louder, std::string const& name,
+    AsyncHelloWorker(bool louder, const std::string* name,
```

Comment:
Question re. string_view / string_ref: these abstractions don't own the memory they're pointing to but are basically typed `const&`s - who owns their memory? The buffer? Then who owns the buffer? Do you expect the buffer to be moved into the worker?